### ISSUE TITLE
SQLAlchemyModelFactory _create should at least flush the session 

### ISSUE THREAD

**Issue by vesauimonen** [rbarrois/factory_boy#81]
SQLAlchemyModelFactory _create
(https://github.com/rbarrois/factory_boy/blob/master/factory/alchemy.py#L45-50) should at least flush(I think flush in this context is better than commit) the SQLAlchemy session to produce the behaviour described in the docs:
```python
>>> session.query(User).all()
[]
>>> UserFactory()
<User: User 1>
>>> session.query(User).all()
[<User: User 1>]
```

**Comment by dgouldin** [comment]
I'm using Flask-SQLAlchemy 2.0, SQLAlchemy 1.0.5, and factory_boy 2.5.2. Autoflush is set to true on my flask_sqlalchemy scoped_session (by default), but I still have to manually call `session.flush()` after creating instances via factories in order to get the model's id to populate. Is this expected? It's certainly not the behavior I'd desire.

**Comment by jeffwidman** [comment]
I just experimented with this a bunch. Using Flask-SQLAlchemy 2.0, SQLAlchemy 1.0.6 and factory_boy 2.5.2

I have a factory that creates categories. The initial category is seeded with no parent, but after that, all other categories are created with a randomly selected ID of another category as their parent. 
`parent = factory.LazyAttribute(lambda p: random.choice(
        gear_models.GearCategory.query.options(load_only("id")).all()).id)`

I observed the following behavior:
1) when first category created from the factory, it has no ID. 
2) When second category created from the factory, it has no ID, _but_ SQLAlchemy has correctly flushed the previous category and allows this second category to pick the previous category. 

So basically SQLAlchemy handled flushing the category when it was required in order to provide a PK ID that another object could use. But it didn't flush it until then... ie, if I tried to access ID immediately upon creation, it failed.

To me, this seems like reasonable default behavior on the part of SQLAlchemy. If for some reason you need a PK ID right away, then it's easy enough to call db.session.flush() as a Factory boy post_generation hook. 

Personally, I think it'd be just fine to close this issue.
